internal/metamorph/callbacker.go

```
@@ -30,16 +40,52 @@ type Callback struct {
 }
 
 type Callbacker struct {
-	httpClient HttpClient
-	wg         sync.WaitGroup
-	mu         sync.Mutex
-	disposed   bool
+	httpClient      HttpClient
+	wg              sync.WaitGroup
+	mu              sync.Mutex
+	disposed        bool
+	callbackerStats *CallbackerStats
 }
 
 func NewCallbacker(httpClient HttpClient) *Callbacker {
```

If there is a possible error in this function, then it should return that error

```
func NewCallbacker(httpClient HttpClient) (*Callbacker, error) {...}
```